rfc-079-script-imports-of-production-data.md

```diff
@@ -0,0 +1,56 @@
+# Script imports of production data
+
+## Summary
+
+Developers writing production data to file on a local or remote server MUST do
+this via an automated script, rather than doing this manually, e.g. via the
+Rails console.
+
+This script MUST anonymise any sensitive data, and the data dump MUST be stored
+outside of the working repository to ensure it can't be inadvertedly committed
+alongside any changes to the codebase.
+
+## Problem
+
+Fetching data from a database by hand is error-prone. A data dump may be left
+on a server, committed to a repository or even in some cases alter production
+data in unforeseen ways. Developers may not realise they have exposed or
+impacted production data until long after an incident occurs.
+
+By writing a script to automate this process, we ensure that every import of
+data from a production environment is done in the same way, minimising room for
+error. It also allows us to scrub the resulting data dump of any
+personally-identifying and security-sensitive information.
+
+Committing a data-fetching script to the repository makes it visible to other
+members of the team in the code review process, and allows others to provide
+feedback to ensure we're reaching a consensus on the safest way of fetching
+this data.
+
+## Proposal
+
+If developing an application requres a developer to write any production data
+containing sensitive data to a local file or a file on a remove server, this
+MUST be done via an automated script.
+
+The script MUST anonymise any personally-identifying or security-sensitive
+data.
```

Comment:
Do we want to specify whether the anonymisation should take place on the server before the data is pulled down? That would be more secure (e.g. avoids the possibility of the script being interrupted part-way through, resulting in someone having pulled the data to a local file but it not being anonymised), but could increase load on the prod db server.

Reply:
I think it would be good for anonymisation to happen before anything is written to the disc. That could still happen locally though, if the data is piped through a script that substitutes values before writing it.

Reply:
Asking for a friend: where can I find a complete list of personally-identifying and security-sensitive data? 
E.g. email addresses, are in those two categories?
Thanks

Reply:
The ICO list is a useful starting point: https://ico.org.uk/for-organisations/guide-to-data-protection/guide-to-the-general-data-protection-regulation-gdpr/key-definitions/what-is-personal-data/#1

Is that something we should link to from this RFC?

Reply:
Seems like a good idea to add a link.

Reply:
Thanks all, have added the link in and specified that anonymisation should take place before writing to the file.